interlude/src/Features/Multiplayer/LobbyScreen.fs

```diff
@@ -20,27 +20,35 @@ type LobbySettingsPage(settings: LobbySettings) as this =
     let host_rotation = Setting.simple settings.HostRotation
     let auto_countdown = Setting.simple settings.AutomaticRoundCountdown
 
-    do
-        page_container()
-        |+ PageTextEntry("lobby.name", name).Pos(0)
-        |+ PageSetting("lobby.host_rotation", Selector<_>.FromBool(host_rotation))
-            .Pos(3)
-            .Tooltip(Tooltip.Info("lobby.host_rotation"))
-        |+ PageSetting("lobby.auto_countdown", Selector<_>.FromBool(auto_countdown))
-            .Pos(5)
-            .Tooltip(Tooltip.Info("lobby.auto_countdown"))
-        |> this.Content
-
-    override this.Title = %"lobby.name"
-
-    override this.OnClose() =
+    let submit () =
         Lobby.settings
             {
                 Name = name.Value
                 HostRotation = host_rotation.Value
                 AutomaticRoundCountdown = auto_countdown.Value
             }
 
+    let submit_button =
+        PageButton (
+            "confirm.yes",
+            (fun () ->
+                submit ()
```

Comment:
> this does have an issue where pressing esc or back in the bottom left will cause the settings to not apply. That should be a simple fix, though I do not know how to go about it.

Because the save only happens when the button is clicked, pressing ESC or clicking Back will not save changes (because only `OnClose()` is called in these cases)

It could work better to always save the user's changes as before, but some validation is done on the name before it gets sent, and if invalid then it sends the old name unchanged.

AND/OR
Handle this server side:
- I need to double check if the server replies with what the settings have now been changed to, or if the client assumes that what they sent is now the settings
- If the server doesn't have final say, update the protocol so it does, probably a task for only me ATM due to the mess the multiplayer code is in
- Once the server has final say, just have some code on the server end to truncate long lobby names, filter out unprintable characters and ignore changes from current name to empty string

I will look into how the protocol works for this and then get back to you

Reply:
Update: Just confirmed the server has the opportunity to validate/edit the saved settings before broadcasting them back to all lobby players

So on line 619 of Lobbies.fs in Interlude.Web.Server we could add validation
Funnily enough I already have a function called `valid_lobby_name` that does this on creation of the lobby, just not on edit
So when saving the settings, if the new name doesn't pass validation then we use the old name instead (silently for now, with something in the UI to tell you what happened later)

Deploying updates to server stuff is currently held back by some pending chart database changes, when I finish those up (could be this weekend) then new changes to the server can be deployed

If you wanna try making changes to the server side and testing it locally yourself, let me know as it takes a few more setup steps than building the client (and could do with a beefy PC if using Docker on Windows)